src/beamlime/constructors/providers.py

```diff
@@ -233,13 +233,17 @@ def __init__(
         from functools import lru_cache
 
         super().__init__(_constructor, *args, **kwargs)
+        self.cached_result: Product
         self.cached_args = CachedArguments()
-        self.cached_constructor = lru_cache(maxsize=1)(self._constructor)
         self.cache_indicator = lru_cache(maxsize=2)(self.cached_args)
 
     def __call__(self, *args: Any, **kwargs: Any) -> Product:
         self.cache_indicator(*self.args, *args, **self.keywords, **kwargs)
-        if self.cache_indicator.cache_info().currsize > 1:
+        if not hasattr(self, "cached_result"):
+            self.cached_result = self.constructor(
```

Comment:
You changed it from `self._constructor` to `self.constructor` without any changes elsewhere. Which one is correct?

Reply:
`self.constructor` is the actual exposed property, so I thought it might be more reasonable to use that one here.
`Provider` doesn't do anything but just gives the `self._constructor` when `self.constructor` is called.